### PR TITLE
Turbo fish assist supports multiple type arguments

### LINKED ISSUE #11135: "Add a turbofish" code action should respect the number of type arguments
There is a nice code action `` Add `::<>` ``  which adds a turbofish `::<_>` between the function identifier and opening parenthesis.  However it actually always add a single placeholder `<_>` instead of what number needed defined in the declaration.  It should add exactly as many underscores as needed.

### PR DESCRIPTION
**Pull request by Vannevelj**
This fixes #11135 (changelog: bug).

I've only started using Rust a few days ago but saw this issue on the top of the list when I looked at this repo. I based myself on [this blog post](https://techblog.tonsser.com/posts/what-is-rusts-turbofish) to understand what a "turbo fish" is so let me know if I missed anything.

### CODE REVIEW

File: crates/ide_assists/src/handlers/add_turbo_fish.rs [hunk 1]
```diff
@@ -77,13 +78,22 @@ pub(crate) fn add_turbo_fish(acc: &mut Assists, ctx: &AssistContext) -> Option<(
         }
     }
 
+    let number_of_arguments = generics.len();
+    let fish_head = std::iter::repeat("_").take(number_of_arguments).collect_vec().join(",");
```
Comment (TonalidadeHidrica) [inline crates/ide_assists/src/handlers/add_turbo_fish.rs:82]: Thanks for very quick implementation!
Just a nitpick: you can use `itertools::repeat_n('_', number_of_arguments).join(",")` for a bit concise code.

Comment (TonalidadeHidrica) [reply crates/ide_assists/src/handlers/add_turbo_fish.rs:82]: But wait, I found a Japanese [article](https://maguro.dev/rust-itertools-join/) that collecting into vec firsthand is faster.  So please just forget, sorry.

**Review by matklad — APPROVED** [review]
LGTM!

I'd personally write this as `.collect::<Vec<_>>` rather than `.collect_vec` to avoid importing itertools (there's also a handy `add_turbo_fish` assist which makes this ergonomic :P ), but that's just a nitpick. 

bors r+

**Comment by lnicola** [comment]
changelog fix (first contribution) use the right numbers of placeholders in `Add turbofish`.

**Pull request merged**